src/nile/core/declare.py

```diff
@@ -42,6 +44,12 @@ def declare(
     logging.info(f"🧾 Transaction hash: {hex(tx_hash)}")
 
     deployments.register_class_hash(class_hash, network, alias)
+
+    if watch_mode is not None:
+        if status(tx_hash, network, watch_mode).status.is_rejected:
+            deployments.unregister(class_hash, network, alias, is_declaration=True)
```

Comment:
this makes me think the default should be `watch_mode=debug`. otherwise it's easy for users to enter an invalid state (failed deployment being registered)

Reply:
yes i think it should

Reply:
Haha I don't disagree on that point. Okay, let's give `watch_mode=debug` a trial run. If it doesn't go over well, we can always change it

Reply:
the alternative is to store the tx_hash. if there's a tx_hash, when someone tries to read the deployment it fetches the status, removing the entry on failure and completing the deployment info on success

Reply:
I don't love the idea of deployment registration being a two-step process for users. I'm not opposed to storing the tx hash (in conjunction with the deployment data) and removing the entry if it's rejected.

Or, what about storing just the hash _with_ `watch_mode=debug`? That will prevent users from sending txs to contracts that haven't been accepted yet (I ran into this error a few times) and, as you mentioned, complete the deployment info on success

Reply:
what don't you like about the two step process? it can be invisible to the user, since the 2nd step will happen on read.

Reply:
On second thought, I take it back. I can see the two-step process being a good option